Client/Frontend/Browser/TabScrollController.swift

```diff
@@ -38,23 +38,38 @@ class TabScrollingController: NSObject {
 
     var footerBottomConstraint: Constraint?
     var headerTopConstraint: Constraint?
-    var toolbarsShowing: Bool { return headerTopOffset == 0 }
+    var toolbarsShowing: Bool { return self.urlBarState == 1.0 }
     private var suppressToolbarHiding: Bool = false
 
-    private var headerTopOffset: CGFloat = 0 {
-        didSet {
-            headerTopConstraint?.updateOffset(headerTopOffset)
-            header?.superview?.setNeedsLayout()
-        }
+    private var minifiedHeaderTopOffset: CGFloat {
+        return URLBarViewUX.MinifiedURLBarHeight - (self.header?.bounds.height ?? 0)
     }
 
-    private var footerBottomOffset: CGFloat = 0 {
-        didSet {
-            footerBottomConstraint?.updateOffset(footerBottomOffset)
-            footer?.superview?.setNeedsLayout()
+    private var urlBarState: CGFloat {
+        get {
+            return self.urlBar?.state ?? 0.0
+        }
+        set {
+            guard let urlBar = self.urlBar else {
+                return
+            }
+            urlBar.state = newValue
+            let inverseState = 1.0 - urlBar.state
+            self.headerTopConstraint?.updateOffset(inverseState * self.minifiedHeaderTopOffset)
+            self.footerBottomConstraint?.updateOffset(inverseState * self.bottomScrollHeight)
+            self.header?.superview?.setNeedsLayout()
+            self.footer?.superview?.setNeedsLayout()
```

Comment:
Not sure if we want to setNeedsLayout on the superview everytime a scroll happens. The superview for these views is the entire BVC which includes the WKWebView. Is there a way to prevent calling layout on the entire VC?